pctasks/dataset/pctasks/dataset/workflow.py

```diff
@@ -282,13 +280,66 @@ def create_process_items_workflow(
     )
 
     if is_update_workflow:
-        if workflow_definition.args is None:
-            workflow_definition = workflow_definition.copy(update={"args": ["since"]})
-        else:
-            workflow_definition.args.append("since")
+        workflow_definition = modify_for_update(workflow_definition)
+
+    return workflow_definition
 
-        for input_ in workflow_definition.jobs["create-splits"].tasks[0].args["inputs"]:
-            input_["chunk_options"]["since"] = "${{ args.since }}"
+
+def modify_for_update(workflow_definition: WorkflowDefinition) -> WorkflowDefinition:
+    """
+    Modify a WorkflowDefinition for an update pipeline.
+
+    Parameters
+    ----------
+    workflow_definition: WorkflowDefinition
+
+    Returns
+    -------
+    WorkflowDefinition
+
+    Notes
+    -----
+    This function will return a new WorkflowDefinition with the following changes:
+
+    1. A ``since`` argument will be added to the workflow's ``args``.
+    2. The `create-splits` task will be modified to set the ``chunk_options.since`` to
+       the literal ``${{ args.since }}``, which will be templated to the actual value
+       at submission time.
+    3. The `create-chunks` task will be modified to include include the ``since``
+       value in the chunk file prefix at ``dst_uri``.
+    4. The ``process-chunk`` task will be modified to include the ``since`` value in
+       the in the chunk file prefix at ``item_chunkset_uri``.
+    """
+    workflow_definition = workflow_definition.copy(deep=True)
+    if workflow_definition.args is None:
+        workflow_definition.args = ["since"]
+    else:
+        workflow_definition.args.append("since")
+
+    # Add a 'since' argument to each create-splits
+    for input_ in workflow_definition.jobs["create-splits"].tasks[0].args["inputs"]:
```

Comment:
Could this fail if `use_existing_chunks` is True?

Reply:
It does fail with a `KeyError`.

I think we should error early in that case, saying that `--is-update-workflow` and `--use-existing-chunks` are incompatible.